calico/felix/actor.py

```diff
@@ -259,6 +293,13 @@ def _step(self):
                 # Most-likely a bug.  Report failure to all callers.
                 _log.exception("_finish_msg_batch failed.")
                 results = [(None, e)] * len(results)
+                for uuid in batch_uuids:
```

Comment:
Better to loop over `batch` instead, I think, since that will give you the messages in order (whereas a set is not consistent in its order).